File: src/main/java/xyz/devcmb/treeTumblers/ui/UIManager.java
package xyz.devcmb.treeTumblers.ui;

import org.bukkit.entity.Player;
import java.util.HashMap;
import java.util.Map;

public class UIManager {
    public static Map<Player, PlayerUIController> playerUIControllers = new HashMap<>();

    public static PlayerUIController playerAdded(Player player) {
        PlayerUIController playerUIController = new PlayerUIController(player);
        playerUIController.registerAllActionBars();

        playerUIControllers.put(player, playerUIController);
        return playerUIController;
    }
}

File: src/main/kotlin/xyz/devcmb/treeTumblers/ui/UIManager.kt
package xyz.devcmb.treeTumblers.ui

import org.bukkit.entity.Player

object UIManager {
    var delegates: MutableMap<Player, PlayerUIDelegate> = HashMap()

    fun PlayerAdded(player: Player) {
        val delegate = PlayerUIDelegate(player)
        delegate.load()
        delegates[player] = delegate
    }

    fun PlayerRemoved(player: Player) {
        val delegate: PlayerUIDelegate? = delegates[player]
        if(delegate == null) return
        delegate.unload()
        delegates.remove(player)
    }
}